### PR TITLE
[DateInput] add unit test for timePrecision

### PR DESCRIPTION
**Pull request by giladgray**
cc @mfedderly 

this will fix the build.

### CODE REVIEW

File: packages/datetime/src/dateInput.tsx [hunk 1]
```diff
@@ -166,13 +166,13 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         const sharedProps: IDatePickerBaseProps = {
             ...this.props,
             onChange: this.handleDateChange,
-            value: this.isMomentValidAndInRange(this.state.value) ? fromMomentToDate(this.state.value) : undefined,
+            value: this.isMomentValidAndInRange(this.state.value) ? fromMomentToDate(this.state.value) : null,
```
Comment (giladgray) [inline packages/datetime/src/dateInput.tsx]: turns out this change was the culprit of test failure.

File: packages/datetime/test/dateInputTests.tsx [hunk 2]
```diff
@@ -37,11 +37,20 @@ describe("<DateInput>", () => {
     });
 
     it("Popover doesn't open if disabled=true", () => {
-        const wrapper = mount(<DateInput disabled />);
-        wrapper.find(InputGroup).simulate("focus");
+        const wrapper = mount(<DateInput disabled={true} />);
+        wrapper.find("input").simulate("focus");
```
Comment (giladgray) [inline packages/datetime/test/dateInputTests.tsx]: this test didn't actually work before, now it does. (and i had to fix line 188 above)

**Comment by cmslewis** [comment]
Will approve pending passing build.

File: packages/datetime/test/dateInputTests.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { Months } from "../src/common/months";
 import { Classes, DateInput, TimePicker, TimePickerPrecision } from "../src/index";
 import * as DateTestUtils from "./common/dateTestUtils";
 
-describe("<DateInput>", () => {
+describe.only("<DateInput>", () => {
```
Comment (cmslewis) [inline packages/datetime/test/dateInputTests.tsx]: be sure to undo this change

Comment (giladgray) [inline packages/datetime/test/dateInputTests.tsx]: big F

**Comment by mfedderly** [comment]
Cool. Thanks!
